### PR TITLE
feat(topology/plugin): add show-labels in get/list node/nodes 

### PR DESCRIPTION
**Pull request by sinhaashish**
**Get node without labels**
```
[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get node node-0-154304
 ID             GRPC ENDPOINT        STATUS
 node-0-154304  95.216.216.11:10124  Online
 ```
**Get node with labels**

```
ashish@vm:~/code/rust/mayastor-extensions(develop)$ ./target/debug/kubectl-mayastor get node node-0-153279 --show-labels
 ID             GRPC ENDPOINT        STATUS  LABELS
 node-0-153279  135.181.89.44:10124  Online  C=D, A=B
 ```

**List nodes with labels**
```
[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get nodes
 ID             GRPC ENDPOINT        STATUS
 node-0-154304  95.216.216.11:10124  Online
 node-1-154304  37.27.6.24:10124     Online
 node-2-154304  37.27.41.173:10124   Online
 ```
**List nodes with labels**
```
[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get nodes --show-labels
 ID             GRPC ENDPOINT        STATUS  LABELS
 node-1-154304  37.27.6.24:10124     Online  P=Q
 node-0-154304  95.216.216.11:10124  Online  C=D, A=B
 node-2-154304  37.27.41.173:10124   Online
```

### CODE REVIEW

File: control-plane/plugin/src/resources/node.rs [hunk 1]
```diff
@@ -105,18 +111,98 @@ impl List for Nodes {
...
 
+/// The NodeDisplayLabels structure is responsible for controlling the display formatting of Node
+/// objects. `#[serde(flatten)]` and `#[serde(skip)]` attributes are used to ensure that when the
+/// object is serialised, only the `inner` object is represented.
+#[derive(Serialize, Debug)]
+pub struct NodeDisplayLabels {
+    #[serde(flatten)]
+    pub inner: Vec<openapi::models::Node>,
+    #[serde(skip)]
+    show_labels: bool,
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/node.rs:139]: Is this needed?
Shouldn't show_labels be always true when we use this type?

File: control-plane/plugin/src/resources/node.rs [hunk 2]
```diff
@@ -105,18 +111,98 @@ impl List for Nodes {
...
+impl GetWithArgs for Node {
     type ID = NodeId;
-    async fn get(id: &Self::ID, output: &utils::OutputFormat) -> PluginResult {
+    type Args = GetNodeArgs;
+    async fn get(id: &Self::ID, args: &Self::Args, output: &utils::OutputFormat) -> PluginResult {
         match RestClient::client().nodes_api().get_node(id).await {
             Ok(node) => {
-                // Print table, json or yaml based on output format.
-                utils::print_table(output, node.into_body());
+                let node_display =
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/node.rs:217]: There's a mix here of using the NodeDisplayLabels and not using it.
IMHO we should either use it only when show_labels true or then always use it and don't need to check args again

File: control-plane/plugin/src/resources/node.rs (hunk 1 above)
Comment (sinhaashish) [reply control-plane/plugin/src/resources/node.rs:139]: no, its true when `--show-labels` is passed

File: control-plane/plugin/src/resources/node.rs (hunk 2 above)
Comment (sinhaashish) [reply control-plane/plugin/src/resources/node.rs:217]: done

File: control-plane/plugin/src/resources/node.rs [hunk 3]
```diff
@@ -29,13 +28,35 @@ enum NodeCordonDrainState {
 pub struct GetNodeArgs {
     /// Id of the node
     node_id: NodeId,
+    /// Show the labels of the node
+    #[clap(global = true, long, default_value = "false")]
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/node.rs]: does it need to be global? Here and on List

File: control-plane/plugin/src/resources/node.rs (hunk 2 above)
Comment (tiagolobocastro) [reply control-plane/plugin/src/resources/node.rs:217]: Great, very clean now :)

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/plugin/src/resources/node.rs (hunk 3 above)
Comment (sinhaashish) [reply control-plane/plugin/src/resources/node.rs]: no, removed it from both the places

**Review by Abhinandan-Purkait — APPROVED** [review]
Awesome 👍🏻

**Comment by sinhaashish** [comment]
bors merge

**Pull request merged**